### PR TITLE
psychometric frontend added. Model created. Data updation working.

### PR DESCRIPTION
**Pull request by prihit**

### CODE REVIEW

File: profiles/models.py [hunk 1]
```diff
@@ -15,6 +15,14 @@ class Profile(BaseModel):
     def __str__(self):
         return self.user.get_full_name()
 
+class psychometry(BaseModel):
```
Comment (chayan007) [inline profiles/models.py:18]: Please follow Pascal case for naming classes in any language

File: base/views.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect
 
 from college.models import College
-
+from profiles.models import psychometry, Profile
```
Comment (chayan007) [inline base/views.py:4]: 1 line gap when different apps.

File: base/views.py [hunk 3]
```diff
@@ -43,3 +43,22 @@ def terms(request):
 
 def privacy(request):
     return render(request, 'v2/pages/public/privacy.html')
+
+def psycho(request):
```
Comment (chayan007) [inline base/views.py:47]: use CBV instead of doing `request.method == 'POST'`

Comment (chayan007) [reply base/views.py:47]: Use `get()` and `post()`

Comment (chayan007) [reply base/views.py:47]: Also make this CBV authentication protected.

```
if not request.user:
   return render(request, 'some/page.html', {'error': 'Login to give the exam.'})
```

File: base/views.py [hunk 4]
```diff
@@ -43,3 +43,22 @@ def terms(request):
...
+    if request.method == 'POST':
+        placement = request.POST['placement']
+        infrastructure = request.POST['infrastructure']
+        academics = request.POST['academics']
+        profile_obj = Profile.objects.get(user = request.user)
+        if  psychometry.objects.get(profile=profile_obj):
```
Comment (chayan007) [inline base/views.py:53]: `existing_psychometric_record = psychometry.objects.get(profile=profile_obj)`

Do not do multiple queries for single thing. This is bad practice.

File: base/views.py [hunk 5]
```diff
@@ -43,3 +43,22 @@ def terms(request):
...
+            psycho_obj = psychometry.objects.get(profile=profile_obj)
+            psycho_obj.infrastructure = infrastructure
+            psycho_obj.academics = academics
+            psycho_obj.placement = placement
+            psycho_obj.save()
+        else:
+            psycho_obj = psychometry.objects.create(profile=profile_obj, infrastructure=infrastructure, academics=academics, placement=placement)
```
Comment (chayan007) [inline base/views.py:60]: Follow lints. Line should not exceed 80 characters. (Tabs and space included)

File: base/views.py [hunk 6]
```diff
@@ -43,3 +43,22 @@ def terms(request):
...
+            psycho_obj.save()
+        return redirect('/')
```
Comment (chayan007) [inline base/views.py:62]: redirect with message that your test was successful or stuff. Think as an user.

File: college/models.py [hunk 7]
```diff
@@ -65,3 +65,7 @@ class CollegeImages(BaseModel):
 
     def __str__(self):
         return self.college.full_name
+
```
Comment (chayan007) [inline college/models.py:68]: remove empty lines. Only 1 empty line is required as per LINT.

**Comment by chayan007** [comment]
I see a lot of migrations for a single purpose.

```
python manage.py migrate <app> zero
<DELETE THE MIGRATIONS>
python manage.py makemigrations <app>
<ADD, COMMIT and PUSH>
```

File: profiles/models.py (hunk 1 above)
Comment (prihit) [reply profiles/models.py:18]: Done

File: base/views.py (hunk 2 above)
Comment (prihit) [reply base/views.py:4]: Done

File: base/views.py (hunk 3 above)
Comment (prihit) [reply base/views.py:47]: Done but last authentication part is not working.

File: base/views.py (hunk 4 above)
Comment (prihit) [reply base/views.py:53]: done

File: base/views.py (hunk 5 above)
Comment (prihit) [reply base/views.py:60]: done

File: base/views.py (hunk 6 above)
Comment (prihit) [reply base/views.py:62]: done

File: base/views.py [hunk 8]
```diff
@@ -43,3 +46,30 @@ def terms(request):
...
+        else:
+            return render(request, 'v2/pages/public/home.html', {'error': 'You need to login in order to give the exam'})
+
+    def post(self, request):
+        placement = request.POST['placement']
+        infrastructure = request.POST['infrastructure']
+        academics = request.POST['academics']
+        profile_obj = Profile.objects.get(user = request.user)
+        psycho_obj = Psychometry.objects.get(profile=profile_obj)
+        if  psycho_obj:
```
Comment (chayan007) [inline base/views.py:64]: Did you test the code ? By default nullable is false. So you might not be able to create psycho object only with profile

File: base/views.py [hunk 9]
```diff
@@ -43,3 +46,30 @@ def terms(request):
 
 def privacy(request):
     return render(request, 'v2/pages/public/privacy.html')
+
+class PsychoView(View): 
+    def get(self, request):
+        print(request.user)
+        if request.user:
```
Comment (chayan007) [inline base/views.py:53]: https://stackoverflow.com/questions/3644902/how-to-check-if-a-user-is-logged-in-how-to-properly-use-user-is-authenticated

File: base/views.py (hunk 8 above)
Comment (chayan007) [reply base/views.py:64]:
```
Psychometry.objects.create(
    profile=profile_obj,
    infrastructure=infrastructure,
    academics=academics,
    placement=placement
)
```

File: base/views.py (hunk 9 above)
Comment (prihit) [reply base/views.py:53]: image

**Review by chayan007 — APPROVED** [review]

**Comment by chayan007** [comment]
`git pull origin main`

Resolve conflicts and merge when ready.

**Pull request merged**